src/index.ts

```diff
@@ -1212,6 +897,44 @@ class NixInstallerAction extends DetSysAction {
     }
   }
 
+  async cleanupNoSystemd(): Promise<void> {
+    if (!this.forceNoSystemd) {
+      // Nothing to do, we didn't use the double fork
+      return;
+    }
+
+    const pidFile = path.join(this.daemonDir, "daemon.pid");
+    const pid = parseInt(await readFile(pidFile, { encoding: "ascii" }));
+    actionsCore.debug(`found daemon pid: ${pid}`);
+    if (!pid) {
+      throw new Error("the daemon did not start successfully");
+    }
+
+    actionsCore.debug(`killing daemon process ${pid}`);
+
+    try {
+      // Repeatedly signal 0 the daemon to test if it is up.
+      // If it exits, kill will raise an exception which breaks us out of this control flow and skips the sigterm.
+      // If it doesn't exit in 30s, we SIGTERM it.
+      for (let i = 0; i < 30 * 10; i++) {
+        process.kill(pid, 0);
+        await setTimeout(100);
+      }
+
+      this.addFact(FACT_SENT_SIGTERM, true);
+      actionsCore.info(`Sending the daemon a SIGTERM`);
+      process.kill(pid, "SIGTERM");
+    } catch {
+      // Perfectly normal to get an exception here, because the process shut down.
+    }
+
+    if (actionsCore.isDebug()) {
+      actionsCore.info("Entire log:");
+      const entireLog = await readFile(path.join(this.daemonDir, "daemon.log"));
```

Comment:
Did you mean to call `readFile` with an encoding here? You're calling `toString` on a `Buffer` instance.
```suggestion
      const entireLog = await readFile(path.join(this.daemonDir, "daemon.log"), "utf-8");
```